string sender
uint32 jointid
---
string jointname
uint32 jointtype
